core/chains/evm/client/pool.go

```diff
@@ -195,36 +196,41 @@ func (p *Pool) ChainID() *big.Int {
 	return p.chainID
 }
 
-func (p *Pool) getRoundRobin() Node {
-	nodes := p.liveNodes()
-	nNodes := len(nodes)
-	if nNodes == 0 {
-		p.logger.Critical("No live RPC nodes available")
-		return &erroringNode{errMsg: fmt.Sprintf("no live nodes available for chain %s", p.chainID.String())}
+func (p *Pool) getBestNode() Node {
+	highestHeadNumber := int64(-1)
+	var node Node
+	for _, n := range p.nodes {
+		latestReceivedBlockNumber := n.LatestReceivedBlockNumber()
+		if n.State() == NodeStateAlive && n.LatestReceivedBlockNumber() > highestHeadNumber {
+			node = n
+			highestHeadNumber = latestReceivedBlockNumber
+		}
```

Comment:
Ah, this is an interesting solution! I was imagining we would track a state for the node and if it gets more than N out of date it would be marked as offline, but this seems like it might be better. This way we can respond very fast and always use the most up-to-date node. I like it.

Reply:
One thing to be aware of is that where previously we were spreading load across nodes, now all the load may be concentrated on one.

Reply:
> One thing to be aware of is that where previously we were spreading load across nodes, now all the load may be concentrated on one.

This is a feature IMHO. Randomly switching nodes invites inconsistent results, particularly when we are only tracking the head number and don't actually know if the blocks are the same.

Reply:
See the other comment from @jmank88 he suggested to stick to the same node as long as possible.